File: src/test_stack.rs
#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_push() {
        let mut stack = Stack::new();
        stack.push(1);
        assert_eq!(stack.size(), 1);
    }

    #[test]
    fn test_pop() {
        let mut stack = Stack::new();
        stack.push(1);
        let value = stack.pop();
        assert_eq!(value, Some(1));
        assert_eq!(stack.size(), 0);
    }

    #[test]
    fn test_is_empty() {
        let mut stack = Stack::new();
        assert_eq!(stack.is_empty(), true);
        stack.push(1);
        assert_eq!(stack.is_empty(), false);
    }

    #[test]
    fn test_pop_empty() {
        let mut stack = Stack::new();
        let value = stack.pop();
        assert_eq!(value, None);
    }
}

File: test/test_stack.py
import unittest

# Assuming the stack implementation is in a file named stack.py
# and has a class named Stack
from stack import Stack

class TestStack(unittest.TestCase):

    def setUp(self):
        self.stack = Stack()

    def test_push(self):
        self.stack.push(1)
        self.assertEqual(self.stack.items, [1])
        self.stack.push(2)
        self.assertEqual(self.stack.items, [1, 2])

    def test_pop(self):
        self.stack.push(1)
        self.stack.push(2)
        self.assertEqual(self.stack.pop(), 2)
        self.assertEqual(self.stack.items, [1])
        self.assertEqual(self.stack.pop(), 1)
        self.assertTrue(self.stack.is_empty())
        with self.assertRaises(IndexError):
            self.stack.pop()

    def test_is_empty(self):
        self.assertTrue(self.stack.is_empty())
        self.stack.push(1)
        self.assertFalse(self.stack.is_empty())

if __name__ == '__main__':
    unittest.main()
